ctapipe_io_magic/__init__.py

```diff
@@ -180,9 +180,10 @@ def __init__(self, input_url=None, config=None, parent=None, **kwargs):
             self.file_list = []
 
             regex = rf"\d{{6}}_M{telescope}_0{run}\.\d{{3}}_Y_.*\.root"
+            regex_mc = rf"GA_M{telescope}_\w+_{run}_Y_.*\.root"
```

Comment:
You can use `regex = re.compile(....)` here, and then directly `regex.findall`. Though I find usage of findall for this case strange here. I would have expected a `match` call and just checking `regex.match(...) is not None`.

You can also use `path.iterdir()` for the loop and then you'll get `pathlib.Path` for the entries and you can just `match` `path.name`.

Reply:
I followed your suggestions here. Also, for some reason I was trying to match real data with only 6 digits at the beginning, not 8. Copy paste from the function below, which is also now fixed, but was working because `findall` was used. But `match` can be used there as well, I will change